tests/unit/test_internal_utils.py

```diff
@@ -101,3 +102,31 @@ def test_generate_random_alphanumeric():
         futures = [executor.submit(generate_random_alphanumeric) for _ in range(5)]
         res = [f.result() for f in futures]
         assert len(set(res)) == 5  # no duplicate string
+
+
+@pytest.mark.parametrize(
+    "string, expected_result",
+    [
+        ('foo.bar."hello.world".baz', ["foo", "bar", '"hello.world"', "baz"]),
+        ('"a.b".c."d.e.f".g', ['"a.b"', "c", '"d.e.f"', "g"]),
+        ('x."y.z".w', ["x", '"y.z"', "w"]),
+        ("noquotes.here", ["noquotes", "here"]),
+        ('"only.one.token"', ['"only.one.token"']),
+        ('start."middle.with.dots".end', ["start", '"middle.with.dots"', "end"]),
+        ('part.with"quote".inside', ["part", 'with"quote"', "inside"]),
+        (
+            '"quoted with ""embedded"" quotes".end',
+            ['"quoted with ""embedded"" quotes"', "end"],
+        ),
+        (
+            '"quoted with ""embedded.and.some.dots"" quotes".end',
+            ['"quoted with ""embedded.and.some.dots"" quotes"', "end"],
+        ),
+        (
+            '"quoted with ""embedded.and.some.dots"" quotes escaped".end',
+            ['"quoted with ""embedded.and.some.dots"" quotes escaped"', "end"],
+        ),
```

Comment:
can we also add a case where we have open quotes
`'"open.quotes.end'`

Reply:
done